cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -359,6 +425,18 @@ const FunctionDoc div_checked_doc{
      "integer overflow is encountered."),
     {"dividend", "divisor"}};
 
+const FunctionDoc pow_doc{
+    "Raise arguments to power element-wise",
+    ("Integer to negative integer power returns an error. However, integer overflow\n"
+     "wraps around. Floating poing power follows std::pow() behaviour.\n"),
```

Comment:
```suggestion
     "wraps around. Floating point power follows std::pow() behaviour.\n"),
```

Reply:
My fault. I updated @rok 's docstring. Will fix the typo.